File: VIBEBackend/AvatarService/Models/AvatarModel.cs
namespace AvatarService.Models
{
	public class AvatarModel
	{
		public long Id { get; set; }
		public string Name { get; set; }
		public virtual ICollection<AnalysisModel> Analysis { get; set; }

		public AvatarDto Convert()
		{
			var dto = new AvatarDto
			{
				Id = Id,
				Name = Name
			};

			List<AnalysisDto>? analysisList = new();
			if (Analysis != null)
				foreach (var item in Analysis)
				{
					analysisList.Add(item.Convert());
				}

			dto.Analysis = analysisList;
			return dto;
		}
	}
}

File: frontend-vibe/src/app/models/avatar.model.ts
import AnalysisModel from "./analysis-model";

export default class AvatarModel {
  id: number | null;
  name: string;
  analysis: AnalysisModel[] = [];

  constructor(
    id: number | null = null,
    name: string | null = null
  ) {
    this.id = id;
    this.name = name;
  }
}
